### PR TITLE
Uses the Search Bar to give dropdown of companies

### LINKED ISSUE #28: Search For Company CIK's
## Describe the Feature

Be able to use the Search Bar with the API to output the company names for the dropdown.

## Expected Behavior of Feature

As you type the dropdown shows the results.

## Feature Location

Will go in the Search Page

## Additional context

N/A

### PR DESCRIPTION
**Pull request by JimtheBurger**
# Proposed Changes

* Uses API for Search Dropdown results

Fixes #28 .

### CODE REVIEW

File: adversarial_apps/src/components/searchbar.tsx [hunk 1]
```diff
@@ -25,11 +26,19 @@ export default function SearchBar({ placeholder }: {placeholder: string}) {
...
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify(data)
+        });
+        const result = await response.json();
+        console.log("Result is for:", result);
+        const companyNames = result.companies.map((company: { "Company Name": string }) => company["Company Name"]);
+        setResults(companyNames)
       // Mock results for demonstration; replace with actual fetch
```
Comment (SavvyDolphin77) [inline adversarial_apps/src/components/searchbar.tsx]: Remove // mock comments for demonstration line since no longer referencing code

**Review by SavvyDolphin77 — CHANGES_REQUESTED** [review]
Looks good, you show cased it worked so trust the logic. Only thing I suggest adding is more comments and the fix I mentioned.

File: adversarial_apps/src/components/searchbar.tsx (hunk 1 above)
Comment (JimtheBurger) [reply adversarial_apps/src/components/searchbar.tsx]: Fixed

**Review by SavvyDolphin77 — APPROVED** [review]
Looks good.

**Pull request merged**